analytics.js

```diff
@@ -355,31 +361,41 @@ class Analytics {
         object.clientIntegrationObjects,
       );
 
-      // send to all integrations now from the 'toBeProcessedByIntegrationArray' replay queue
-      for (let i = 0; i < succesfulLoadedIntersectClientSuppliedIntegrations.length; i += 1) {
-        try {
-          if (
-            !succesfulLoadedIntersectClientSuppliedIntegrations[i].isFailed ||
-            !succesfulLoadedIntersectClientSuppliedIntegrations[i].isFailed()
-          ) {
-            if (succesfulLoadedIntersectClientSuppliedIntegrations[i][methodName]) {
-              const sendEvent = !object.IsEventBlackListed(
-                event[0].message.event,
-                succesfulLoadedIntersectClientSuppliedIntegrations[i].name,
-              );
-
-              // Block the event if it is blacklisted for the device-mode destination
-              if (sendEvent) {
-                const clonedBufferEvent = cloneDeep(event);
-                succesfulLoadedIntersectClientSuppliedIntegrations[i][methodName](
-                  ...clonedBufferEvent,
-                );
-              }
-            }
-          }
-        } catch (error) {
-          handleError(error);
+      const intgWithoutTransformation = [];
+      const intgWithTransformation = [];
+
+      // Depending on transformation is connected or not
+      // create two sets of destinations
+      succesfulLoadedIntersectClientSuppliedIntegrations.forEach((intg) => {
+        if (intg.areTransformationsConnected) {
+          intgWithTransformation.push(intg);
+        } else {
+          intgWithoutTransformation.push(intg);
         }
+      });
+
+      // loop through destinations that doesn't have
+      // transformation connected with it and send events from the 'toBeProcessedByIntegrationArray' replay queue
+      intgWithoutTransformation.forEach((intg) => {
+        this.sendDataToDestination(intg, event[0], methodName);
+      });
+
+      if (intgWithTransformation.length) {
+        // Process Transformation
+        processTransformation(event[0], this.writeKey, (transformedPayload) => {
+          if (transformedPayload) {
+            intgWithTransformation.forEach((intg) => {
+              // filter the transformed event for that destination
+              const transformedEvents = transformedPayload.filter(
+                (e) => e.destination.id === intg.destinationId,
+              )[0].destination.payload;
```

Comment:
This doesn't seem to align with the response format.

Reply:
Yes, it aligns with the response.

Reply:
https://www.notion.so/rudderstacks/Technical-Design-Doc-018f351a2ed84aa585d49a2e61018304#5fea49acefd64356be1d8bca5759712f

Reply:
Addressed